This is an analysis of Chocolate sales for Ages chocolate stores for the year 2022-2023

In [50]:
#import the relevant libraries

import os

from dotenv import load_dotenv

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


In [51]:
#Loading the dataset

load_dotenv()

path = os.getenv('DATA-PATH')

df = pd.read_csv(path)

In [52]:
# Overview of the dataset

df.info()

df.describe()

df.shape

df.head(10)

<class 'pandas.DataFrame'>
RangeIndex: 200000 entries, 0 to 199999
Data columns (total 11 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   Order_ID         200000 non-null  str    
 1   Product          200000 non-null  str    
 2   Country          200000 non-null  str    
 3   Channel          200000 non-null  str    
 4   Salesperson      200000 non-null  str    
 5   Order_Date       199563 non-null  str    
 6   Discount_Pct     199511 non-null  float64
 7   Price_per_Box    199543 non-null  float64
 8   Marketing_Spend  199539 non-null  float64
 9   Boxes_Shipped    200000 non-null  int64  
 10  Amount           200000 non-null  str    
dtypes: float64(3), int64(1), str(7)
memory usage: 29.6 MB


,Order_ID,Product,Country,Channel,Salesperson,Order_Date,Discount_Pct,Price_per_Box,Marketing_Spend,Boxes_Shipped,Amount
0,ORD-069833,Truffle Gift Box,Australia,Retail,Arjun Mehta,2022-12-11,3.5,13.72,202.03,71,912.31
1,ORD-090726,85% Dark Bar,Australia,Retail,Arjun Mehta,2023-03-14,9.4,3.30,55.18,84,245.91
2,ORD-042159,70% Dark Bar,Japan,Retail,Hannah Müller,2023-12-21,4.9,18.21,60.65,35,583.7
3,ORD-197166,Hazelnut Milk Bar,Germany,Retail,Arjun Mehta,2023-12-18,15.0,2.66,52.00,92,211.27
4,ORD-112162,Almond Crunch Bar,Australia,Retail,Yuki Sato,2023-08-18,4.4,2.75,187.44,214,549.69
5,ORD-051262,70% Dark Bar,Brazil,Online,Lucas Walker,2023-08-04,17.8,12.87,76.94,47,527.01
6,ORD-041587,Hazelnut Milk Bar,Australia,Wholesale,Ananya Gupta,2023-02-20,12.0,3.18,241.18,159,470.63
7,ORD-017541,Mixed Assortment Box,Australia,Online,Lucas Walker,2022-04-01,22.6,2.43,94.27,326,629.51
8,ORD-134320,85% Dark Bar,Germany,Retail,Rohan Patel,2022-08-26,13.9,2.98,68.88,70,181.16
9,ORD-068442,Truffle Gift Box,Germany,Retail,Priya Sharma,2022-01-01,16.3,3.13,102.50,213,588.18


In [53]:
## checking for duplicated or null values

df.isnull().sum()

Order_ID             0
Product              0
Country              0
Channel              0
Salesperson          0
Order_Date         437
Discount_Pct       489
Price_per_Box      457
Marketing_Spend    461
Boxes_Shipped        0
Amount               0
dtype: int64

fix the null coloumns, the empty coulmns will be dropped for the sake of tiume series analysis(with regards date)

In [54]:
df = df.dropna()

In [55]:
df.isnull().sum()

Order_ID           0
Product            0
Country            0
Channel            0
Salesperson        0
Order_Date         0
Discount_Pct       0
Price_per_Box      0
Marketing_Spend    0
Boxes_Shipped      0
Amount             0
dtype: int64

In [56]:
df.duplicated().sum()

np.int64(0)

the dataset indicates a lack of duplicated values

In [ ]:
## fixing the data type of the amount column

df.Amount = df.Amount.astype('float64')

df.Amount.sum()


np.float64(101746622.98)

## Product performance

In [69]:
## revenue by product

rev_product = df.Amount.groupby(df['Product']).sum()

rev_product.sort_values(ascending=False)

Product
70% Dark Bar            26159678.68
Mixed Assortment Box    19405032.16
Truffle Gift Box        19237789.55
Milk Classic Bar         6196107.09
Almond Crunch Bar        5732351.02
Pralines Gift Box        5708907.99
Choco Coins Bag          4739414.39
Mint Dark Bar            3488813.73
White Choco Bar          3196881.88
85% Dark Bar             2944074.50
Salted Caramel Bar       2782369.73
Hazelnut Milk Bar        2155202.26
Name: Amount, dtype: float64

In [70]:
## Shipping by product

shipped_product = df.Boxes_Shipped.groupby(df['Product']).sum()

shipped_product.sort_values(ascending=False)

Product
70% Dark Bar            7118858
Mixed Assortment Box    5299826
Truffle Gift Box        5222685
Milk Classic Bar        1681183
Almond Crunch Bar       1567640
Pralines Gift Box       1564888
Choco Coins Bag         1277879
Mint Dark Bar            949128
White Choco Bar          872678
85% Dark Bar             796875
Salted Caramel Bar       746017
Hazelnut Milk Bar        593618
Name: Boxes_Shipped, dtype: int64

The revenue sum shows "70% dark bar" as the best performing product and the "Mixed assortment box" a close second :

#### 70% dark bar = $26,159,678.68 and total shipped boxes = 7,118,858 boxes
#### Mixed asssortment box =  $19,405,032.16 and total shipped boxed = 5,299,826 boxes 